src/Orchard/Localization/Services/DefaultLocalizedStringManager.cs

```diff
@@ -28,13 +31,15 @@ public class DefaultLocalizedStringManager : ILocalizedStringManager {
             ICacheManager cacheManager,
             ILocalizationStreamParser locationStreamParser,
             ShellSettings shellSettings,
-            ISignals signals) {
+            ISignals signals,
+            ShellDescriptor shellDescriptor) {
```

Comment:
Are you sure the `ShellDescriptor` only contains the enabled features? There is also `IFeatureManager` which contains a `GetEnabledFeatures`. Also there is a mismatch between enabled features and localization files are these files don't mention features per se, just the module, but I am not sure you can do anything about it.